src/bin/console/console.cpp

```diff
@@ -455,57 +472,72 @@ int Console::_generate_tpch(const std::string& args) {
-  const std::string& extension = file_parts.back();
+  const auto tablename = arguments.size() >= 2 ? arguments[1] : std::string{filepath.stem()};
 
-  out("Loading " + filepath + " into table \"" + tablename + "\" ...\n");
+  out("Loading " + std::string(filepath) + " into table \"" + tablename + "\"\n");
 
   auto& storage_manager = StorageManager::get();
   if (storage_manager.has_table(tablename)) {
     storage_manager.drop_table(tablename);
     out("Table " + tablename + " already existed. Replacing it.\n");
   }
 
-  if (extension == "csv") {
+  if (extension == ".csv") {
     auto importer = std::make_shared<ImportCsv>(filepath, Chunk::DEFAULT_SIZE, tablename);
     try {
       importer->execute();
     } catch (const std::exception& exception) {
-      out("Exception thrown while importing CSV:\n  " + std::string(exception.what()) + "\n");
+      out("Error: Exception thrown while importing CSV:\n  " + std::string(exception.what()) + "\n");
       return ReturnCode::Error;
     }
-  } else if (extension == "tbl") {
+  } else if (extension == ".tbl") {
     try {
       auto table = load_table(filepath);
 
       StorageManager::get().add_table(tablename, table);
     } catch (const std::exception& exception) {
-      out("Exception thrown while importing TBL:\n  " + std::string(exception.what()) + "\n");
+      out("Error: Exception thrown while importing TBL:\n  " + std::string(exception.what()) + "\n");
       return ReturnCode::Error;
     }
-  } else if (extension == "bin") {
+  } else if (extension == ".bin") {
     auto importer = std::make_shared<ImportBinary>(filepath, tablename);
     try {
       importer->execute();
     } catch (const std::exception& exception) {
-      out("Exception thrown while importing binary file:\n  " + std::string(exception.what()) + "\n");
+      out("Error: Exception thrown while importing binary file:\n  " + std::string(exception.what()) + "\n");
       return ReturnCode::Error;
     }
   } else {
     out("Error: Unsupported file extension '" + extension + "'\n");
     return ReturnCode::Error;
   }
 
+  const std::string encoding = arguments.size() == 3 ? arguments[2] : "Unencoded";
+
+  const auto type = encoding_type_to_string.right.find(encoding);
+  if (type == encoding_type_to_string.right.end()) {
+    const auto encoding_options = boost::algorithm::join(
+        encoding_type_to_string.right | boost::adaptors::transformed([](auto it) { return it.first; }), ", ");
+    out("Error: Invalid encoding type: '" + encoding + "', try one of these: " + encoding_options + "\n");
+    return ReturnCode::Error;
+  }
+
+  out("Encoding \"" + tablename + "\" using " + encoding + "\n");
+  try {
+    ChunkEncoder::encode_all_chunks(StorageManager::get().get_table(tablename), type->second);
+  } catch (const std::exception& exception) {
+    out("Exception while encoding - table left partially unencoded:\n  " + std::string(exception.what()) + "\n");
```

Comment:
I thought we agreed that `Assert` failure means "Hyrise is borked, shutting down"?

Can you check whether the encoding options are valid for this table - and not do anything if they are not. I know this is cumbersome, but I really do not want to legitimize the catching of Exceptions from Asserts.

Reply:
(Should use the csv parser directly) + Make the CsvParser use AssertInput.

Reply:
Had my look, beheld greatness.

```
  try {
    gt->execute();
  } catch (const std::exception& exception) {
    out("Exception thrown while loading table:\n  " + std::string(exception.what()) + "\n");
    return ReturnCode::Error;
  }
```

Reply:
The trycatchs in console.cpp range from two catching InvalidInputs (which is okay) over some catching std::exception (which is not okay) that do *probably* not touch global data (hopefully) over those catching std::exception while happily accessing global data (StorageManager::add_table generates stats, e.g.) and potentially turning hyrise into Frankensteins monster. 
This is not okay.

Reply:
Changed it to `encoding_supports_data_type`. Let's have the rest of that discussion in the appropriate issue.